src/System.IO.FileSystem/src/System/IO/FileStatus.Unix.cs

```diff
@@ -113,6 +113,20 @@ public void SetAttributes(string path, FileAttributes attributes)
             if (!_exists)
                 FileSystemInfo.ThrowNotFound(path);
 
+            if (Interop.Sys.CanSetHiddenFlag)
+            {
+                if ((attributes & FileAttributes.Hidden) != 0 && (_fileStatus.UserFlags & (uint)Interop.Sys.UserFlags.UF_HIDDEN) != (uint)Interop.Sys.UserFlags.UF_HIDDEN)
+                {
+                    // If Hidden flag is set and cached file status does not have the flag set then set it
+                    Interop.Sys.LChflags(path, (uint)Interop.Sys.UserFlags.UF_HIDDEN);
+                }
+                else if ((attributes & FileAttributes.Hidden) == 0 && (_fileStatus.UserFlags & (uint)Interop.Sys.UserFlags.UF_HIDDEN) == (uint)Interop.Sys.UserFlags.UF_HIDDEN)
```

Comment:
Seems like this would be better structured as:
```C#
if ((attributes & FileAttributes.Hidden) != 0)
{
    if (_fileStatus.UserFlags & (uint)Interop.Sys.UserFlags.UF_HIDDEN) != (uint)Interop.Sys.UserFlags.UF_HIDDEN)
    {
        ...
    }
}
else
{
    if (_fileStatus.UserFlags & (uint)Interop.Sys.UserFlags.UF_HIDDEN) == (uint)Interop.Sys.UserFlags.UF_HIDDEN)
    {
    }
}
```
otherwise we're unnecessarily doing the attributes & Hidden test again for the else block if the if condition is true in the first clause but false in the second.  Plus it avoids needing to write the same condition check (one inverted) twice.